MapboxCoreNavigation/OfflineDirections.swift

```diff
@@ -196,6 +203,13 @@ public class NavigationDirections: Directions {
         
         NavigationDirectionsConstants.offlineSerialQueue.async { [weak self] in
             guard let result = self?.navigator.getRouteForDirectionsUri(url.absoluteString) else {
```

Comment:
Per https://github.com/mapbox/mapbox-navigation-ios/issues/2273#issuecomment-594084579, the only error case we should handle up here would be that `self` (NavigationDirections) has been deallocated.

Upon closer inspection, I’m wondering if it’s even appropriate to call the completion handler in this case. For mapbox/mapbox-gl-native-ios#210 https://github.com/mapbox/mapbox-gl-native-ios/issues/200#issuecomment-598458116, we found that calling the completion handler after `self` has gone away forces the caller to consider whether `self` has gone away because the parent view controller has gone away or the application is terminating.

In other words, when an application calls this method from a MapViewController that holds a strong reference to NavigationDirections, does it expect the completion handler to magically be called even if that MapViewController has been deallocated (for example, because the user has exited that map screen to go to a different part of the application)? If the application has neglected to make `self` weak inside the completion handler, that could create other unexpected behavior in the application. The completion handler would have to explicitly handle the `.cancelled` case, so at least we would have to document that postcondition on the method, which is unfortunate.

Reply:
Not calling a completion under some conditions may also lead to unexpected behavior. At least such use case must also be documented so that users are aware that completion call is not guaranteed.
Also, if we choose not to rise an explicit error message when `NavigationDirections` were deallocated, `noData` error becomes unused, so we can safely remove it